app/src/main/java/protect/card_locker/LoyaltyCardEditActivity.java

```diff
@@ -787,18 +792,17 @@ public void onResume() {
                 if (!mIconUnsaved && !croppedIcon() && !mIconRemoved) {
                     setCardImage(thumbnail, Utils.retrieveCardImage(this, tempLoyaltyCard.id, ImageLocationType.icon), false);
                 }
-            } else {
-                setTitle(R.string.addCardTitle);
             }
-
-            if (mFrontImageUnsaved && !croppedFrontImage()) {
-                setCardImage(cardImageFront, Utils.loadTempImage(this, TEMP_UNSAVED_FRONT_IMAGE_NAME), true);
-            }
-            if (mBackImageUnsaved && !croppedBackImage()) {
-                setCardImage(cardImageBack, Utils.loadTempImage(this, TEMP_UNSAVED_BACK_IMAGE_NAME), true);
-            }
-            if (mIconUnsaved && !croppedIcon()) {
-                setCardImage(thumbnail, Utils.loadTempImage(this, TEMP_UNSAVED_ICON_NAME), false);
+            else {
+                if (mFrontImageUnsaved && !croppedFrontImage()) {
+                    setCardImage(cardImageFront, Utils.loadTempImage(this, TEMP_UNSAVED_FRONT_IMAGE_NAME), true);
+                }
+                if (mBackImageUnsaved && !croppedBackImage()) {
+                    setCardImage(cardImageBack, Utils.loadTempImage(this, TEMP_UNSAVED_BACK_IMAGE_NAME), true);
+                }
+                if (mIconUnsaved && !croppedIcon()) {
+                    setCardImage(thumbnail, Utils.loadTempImage(this, TEMP_UNSAVED_ICON_NAME), false);
+                }
```

Comment:
You put them in an else for `if (updateLoyaltyCard || duplicateFromLoyaltyCardId) {`, but they shouldn't be.

Please compare the structure with how it looks in the main branch: https://github.com/CatimaLoyalty/Android/blob/ed755c35b4c44fe35638fa9b3e990b48828ea764/app/src/main/java/protect/card_locker/LoyaltyCardEditActivity.java#L777-L803

Reply:
I'll be honest, I have no clue anymore why the original code was set up this way, but you changed it from how it was without an explanation why so it should either not be changed or explained :)

Reply:
Well being honest I had a little mess in my head with all this but whatever.

In the new commit i added the original code but i modified some lines:

https://github.com/Jasielprogramador/Android/blob/17d9ba226c73c108eac0b21801866119aa27623c/app/src/main/java/protect/card_locker/LoyaltyCardEditActivity.java#L780

i added this because if u don't the photos that where saved in the original card won't be copied in the duplicated one. This is because in the case of duplication you don't put a value for the update (https://github.com/Jasielprogramador/Android/blob/17d9ba226c73c108eac0b21801866119aa27623c/app/src/main/java/protect/card_locker/LoyaltyCardViewActivity.java#L743) as you can see you only put values in the id and in the duplicateId fields.

So if u don't modify the code of the `LoyaltyCardEditActivity` the photos won't be copied into the duplicated one.

Reply:
You undid the correct title logic again, but I'll just fix that manually :)